universal/include/userver/utils/macaddr/macaddr_utils.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <array>
+#include <stdexcept>
+#include <userver/utils/encoding/hex.hpp>
+#include <utility>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace utils::macaddr::details {
+
+template <std::size_t N>
+class Macaddr final {
```

Comment:
If members of this type (e.g. the constructor and `octets`) should be touched by the user, then this type should not be in `details` or `impl`